### PR TITLE
Create repo_process_guidelines.md

### PR DESCRIPTION
**Pull request by bumblefudge**
Uploading guidance adapted from DIDComm WG precedents for contributors and peer-reviewers coming in blind from outside the working group.  See issue #78

### CODE REVIEW

**Comment by bumblefudge** [comment]
NB @wyc ?

**Comment by wyc** [comment]
@bumblefudge I think this is a good direction. What could help newcomers even more is understanding the review and approval process. Perhaps we can add to the README.md something similar to the statement in [did-core](https://github.com/w3c/did-core#contributing-to-the-repository) to make it explicit how to get new changes in:

> Use the standard fork, branch, and pull request workflow to propose changes to the specification. Please make branch names informative—by including the issue or bug number for example.
>
> Editorial changes that improve the readability of the spec or correct spelling or grammatical mistakes are welcome.
>
> Please read CONTRIBUTING.md, about licensing contributions.

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+### Repository Maintainer selection
+Maintainers must be Working Group Participants selected by Working Group Participants via consensus. At 	least 3 maintainers must be to be active at any given time operating the Repository Process for the Working Group. Maintainers *MUST* have different organizational affiliations to represent the diversity of the Working Group (i.e., no two maintainers should work for the same company, organization, or project).
+
+### Specification Formatting Requirements
+
+* 80-char fixed-width columns (aka "hard wraps") required by this specification.
+
+### Repository process
+
+The Repository process is an extension of the *Deliverable Development Process – Draft* section detailed in DIF’s Charter. The goal of this process is to introduce a repository management processes for the work section developed by the Working Group.
```
Comment (brentzundel) [inline contributing.md:25]:
```suggestion
The Repository process is an extension of the *Deliverable Development
Process – Draft* section detailed in DIF’s Charter. The goal of this process
is to introduce a repository management processes for the work section
developed by the Working Group. 
```
In light of the recommended formatting requirements, should this document be hard wrapped at 80 characters?

**Comment by nembal** [comment]
https://github.com/decentralized-identity/presentation-exchange/issues/78 -- I left comments on this issue, what about those?

**Comment by bumblefudge** [comment]
> #78 -- I left comments on this issue, what about those?

@nembal , sorry, I misunderstood.  I went ahead and addressed all the minor ones, as those seemed uncontroversial and useful to this work item.  As for versioning, I think that would require more discussion and delay.  I am, however, more than willing to add that section to the template for other work items in discussion with the governance group, and add that back into this document as a separate PR in a week or two, after the urgency has passed :D

**Comment by nembal** [comment]
Fair enough. I just wanted to make sure my previous comments are not lost
and will make it to a version.


--
Balázs

**Comment by bumblefudge** [comment]
i can haz merge?

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+## Repository Process
+
+### Repository Maintainer 
+Maintainers are expected to handle the repository management as described in the
+Repository Process (below).
+
+### Repository Maintainer selection
+Maintainers must be Working Group Participants selected by Working Group 
+Participants via consensus. At 	least 3 maintainers must be to be active at any
```
Comment (kdenhartog) [inline contributing.md:20]:
```suggestion
Participants via consensus. At least 3 maintainers must be to be active at any 
```

**Review by kdenhartog — DISMISSED** [review]

**Comment by csuwildcat** [comment]
Homework: read this and get any changes in by Thursday

File: contributing.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+During periods of **normal mode** development, issue-linked PRs should allow up 
+to 5 business day for review and change requests.  During periods of **crunch 
+mode** development, this window is reduced to 2 business days. At least two 
+maintainers should still review all PRs.
+
+
+#### Assigning section content in the STABLE state: 
+
+when all of the following conditions are met:
```
Comment (glcohen) [inline contributing.md:95]: When

File: contributing.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+#### Assigning section content in the REFINING state: 
+
+When the Working Group Participants come to a consensus about the amount and 
+type of content that’s appropriate. While in this state, all substantive changes 
+to the section must be associated with a GitHub issue and must either be 
+approved by a maintainer different from the person who raised the issue, or by 
+consensus among Working Group Participants during a Working Group call. During 
+the REFINING state, the Working Group will request feedback from other 
+decentralized identity-related and -adjacent communities. Ideally, these inputs
+will lead to issues for consensual/deliberative resolution. This merges Further
```
Comment (brentzundel) [inline contributing.md:83]:
```suggestion
will lead to issues for consensual/deliberative resolution. Further
```

File: contributing.md (hunk 3 above)
Comment (brentzundel) [inline contributing.md:95]:
```suggestion
When all of the following conditions are met: 
```

File: contributing.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+#### Assigning section content in the STABLE state: 
+
+when all of the following conditions are met: 
+* There is quality code that demonstrates the concepts it describes.
+* There are no outstanding issues related to that section that the Working Group 
+Participants consider substantive.
+* The Working Group has at least 2 conformant implementations, at least 1 of
+which originated outside of the Working Group 
+
+When consensus has been arrived at that the above conditions have been reached,
```
Comment (brentzundel) [inline contributing.md:102]:
```suggestion
When the WG has consensus that the above conditions have been reached, 
```

File: contributing.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+Participants consider substantive.
+* The Working Group has at least 2 conformant implementations, at least 1 of
+which originated outside of the Working Group 
+
+When consensus has been arrived at that the above conditions have been reached, 
+the section enters a final 2-week comment period has elapsed without any 
+substantive objections to the LAST-COMMENTS status. The start of this period, 
+intended for "last comments," should be announced on the Working Group mailing 
+list. If substantial discussion or change happens, this LAST-COMMENTS phase can
+be extended 1 additional week at the discretion of the chairs.
```
Comment (brentzundel) [inline contributing.md:107]:
```suggestion
the section enters a final 2-week comment period if no substantive
objections are raised against the LAST-COMMENTS status. The start
of this period, intended for "last comments," should be announced
on the Working Group mailing list. If substantial discussion or change
happens, this LAST-COMMENTS phase can be extended 1 additional
week at the discretion of the chairs.
```

Not sure if this wording captures the intended meaning, but the original text wasn't clear to me.

File: contributing.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+Once a section reaches STABLE status, non-substantive changes can be merged at 
+maintainer discretion, however, substantive changes are generally discouraged 
+except by discretion of the chairs. Even then, substantial changes should be 
+discussed on a Working Group call or discussed (however briefly) on a Working 
+Group call if resolved asynchronously via GitHub issues.
```
Comment (brentzundel) [inline contributing.md:113]:
```suggestion
discussed on a Working Group call. If resolved asynchronously via GitHub issues,
the discussion during a Working Group call may be brief.
```

**Review by glcohen — DISMISSED** [review]

**Comment by brentzundel** [comment]
@bumblefudge I think this may be getting ready to merge. Can you look at some of the feedback?

**Comment by bumblefudge** [comment]
Sorry, missed this.  All good!

**Review by glcohen — APPROVED** [review]

**Pull request merged**